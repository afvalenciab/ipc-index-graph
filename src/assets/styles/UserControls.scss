.user--controls {
  display: flex;
  flex-direction: column;
  margin: 0 50px;
  box-sizing: border-box;

  .user--controls_table {
    display: grid;
    grid-template: 50px minmax(auto, 1fr) / 1fr;
    background: #fff;
    border: 1px solid rgba(0,0,0,.1);
    border-radius: 10px;
    padding: 10px;
    box-sizing: border-box;
    height: calc(100vh - 202px);
    min-height: 143px;

    .user--controls_header {
      display: grid;
      grid-template-columns: 3fr 2fr 1fr 1fr;
      grid-gap: 10px;
      border-bottom: 1px solid rgba(87, 87, 87,.7);
      box-sizing: border-box;
      align-items: center;
      font-weight: 700;
      p {
        margin: 0;
      }
    }

    .user--controls_list {
      display: flex;
      flex-direction: column;
      overflow-y: auto;
    }


  }
}