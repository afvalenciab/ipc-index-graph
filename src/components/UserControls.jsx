import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { setUsersListRequest } from '../actions';
import UserItem from './UserItem';
import '../assets/styles/UserControls.scss';

const UserControls = (props) => {
  const { usersList, user, setUsersListRequest } = props;

  useEffect(() => {
    if (user.isAdmin) {
      setUsersListRequest(user);
    }
  }, []);

  return (
    <section className='user--controls'>
      <h2>Administración de Usuarios</h2>
      <div className='user--controls_table'>
        <div className='user--controls_header'>
          <p>Correo electrónico</p>
          <p>Ingresos fallidos</p>
          <p>Admon</p>
          <p>Eliminar</p>
        </div>
        <div className='user--controls_list'>
          {usersList.length > 0 && (
            usersList.map((item) => <UserItem key={item._id} {...item} />)
          )}
        </div>
      </div>
    </section>
  );
};

const mapStateToProps = (state) => {
  return {
    usersList: state.usersList,
    user: state.user,
  };
};

const mapDispatchToProps = {
  setUsersListRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(UserControls);
