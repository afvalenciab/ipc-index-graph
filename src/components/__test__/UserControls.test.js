import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMocks';
import UserControls from '../UserControls';

describe('UserControls testing', () => {
  test('Match Snapshot', () => {
    const userControls = create(
      <ProviderMock>
        <UserControls />
      </ProviderMock>
    );
    expect(userControls.toJSON()).toMatchSnapshot();
  });
});
