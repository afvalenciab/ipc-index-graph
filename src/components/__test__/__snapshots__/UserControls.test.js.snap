// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserControls testing Match Snapshot 1`] = `
<section
  className="user--controls"
>
  <h2>
    Administración de Usuarios
  </h2>
  <div
    className="user--controls_table"
  >
    <div
      className="user--controls_header"
    >
      <p>
        Correo electrónico
      </p>
      <p>
        Ingresos fallidos
      </p>
      <p>
        Admon
      </p>
      <p>
        Eliminar
      </p>
    </div>
    <div
      className="user--controls_list"
    />
  </div>
</section>
`;
